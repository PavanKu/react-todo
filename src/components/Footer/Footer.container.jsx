import React, { useMemo } from "react";

import { useTodos } from "../Todo";
import Footer from "./Footer";
import {
  StyledClearBtnContainer,
  StyledRemainingContainer,
} from "./Footer.styled";
import { RemainingTask } from "../RemainingTask";
import { Tabs } from "../Tabs";
import { ClearBtn } from "../ClearBtn";

import { TAB_OPTIONS } from "../../constants/appConstants";

function FooterContainer() {
  const tabOptions = [
    TAB_OPTIONS.ALL,
    TAB_OPTIONS.ACTIVE,
    TAB_OPTIONS.COMPLETED,
  ];
  const { getTodos, getActiveTab, removeTodos, filterByStatus } = useTodos();
  const todos = getTodos();
  const activeTab = getActiveTab();

  const showFooter = useMemo(() => !!todos.length, [todos]);
  const remainingTaskCount = useMemo(
    () => todos.filter((todo) => !todo.isCompleted).length,
    [todos]
  );
  const completedTaskIds = useMemo(
    () => todos.filter((todo) => todo.isCompleted).map((todo) => todo.id),
    [todos]
  );
  const shouldShowClearBtn = !!completedTaskIds.length;

  return (
    <Footer show={showFooter}>
      <StyledRemainingContainer>
        <RemainingTask count={remainingTaskCount} />
      </StyledRemainingContainer>
      <Tabs
        handleClick={(value) => filterByStatus(value)}
        options={tabOptions}
        active={activeTab}
      />
      <StyledClearBtnContainer show={!!shouldShowClearBtn}>
        <ClearBtn
          label="Clear Completed"
          handleClear={() => removeTodos(completedTaskIds)}
        />
      </StyledClearBtnContainer>
    </Footer>
  );
}

export default FooterContainer;
