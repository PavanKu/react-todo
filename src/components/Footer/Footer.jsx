import React from "react";
import PropTypes from "prop-types";

import StyledFooter from "./Footer.styled";

const DEFAULT_FOOTER_SHOW = false;

function Footer({ show = DEFAULT_FOOTER_SHOW, children }) {
  if (!show) return null;
  return <StyledFooter>{children}</StyledFooter>;
}

Footer.propTypes = {
  show: PropTypes.bool,
};

export default Footer;
