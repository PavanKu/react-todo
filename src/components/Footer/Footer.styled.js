import styled, { css } from "styled-components";

const StyledFooter = styled.footer`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px 15px;
  background: #fefefe;
`;

export const StyledRemainingContainer = styled.div`
  flex: 1;
  background: #fff;
`;

export const StyledClearBtnContainer = styled.div`
  background: #fff;
  margin-left: 30px;
  ${(props) =>
    props.show &&
    css`
      visibility: visible;
    `}
  ${(props) =>
    !props.show &&
    css`
      visibility: hidden;
    `}
`;

export default StyledFooter;
