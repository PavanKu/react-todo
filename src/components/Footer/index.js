import FooterContainer from "./Footer.container";

export { FooterContainer };
