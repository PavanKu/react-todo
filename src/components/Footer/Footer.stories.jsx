import React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs, boolean } from "@storybook/addon-knobs";
import Footer from "./Footer";

const DefaultFooter = () => <Footer show />;
const Playgroud = () => <Footer show={boolean("show", true)} />;

storiesOf("Footer", module)
  .addDecorator(withKnobs)
  .add("Default Footer", DefaultFooter)
  .add("Playground", Playgroud);
