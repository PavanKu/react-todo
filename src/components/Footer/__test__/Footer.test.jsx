import React from "react";
import { cleanup, render } from "@testing-library/react";

import Footer from "../Footer";

describe("<Footer />", () => {
  afterEach(cleanup);

  it("should render without crash", () => {
    render(<Footer show />);
  });

  it("should hide footer based on props", () => {
    const { container } = render(<Footer />);

    expect(container.firstChild).toBeNull();
  });

  it("should match snapshot", () => {
    const { container } = render(<Footer show />);

    expect(container).toMatchSnapshot();
  });
});
