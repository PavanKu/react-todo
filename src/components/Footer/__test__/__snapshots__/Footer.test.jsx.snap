// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<div>
  <footer
    class="sc-AxjAm bzAscD"
  />
</div>
`;
